src/index.js

```diff
@@ -1,13 +1,143 @@
 
-// An example of how you import jQuery into a JS file if you use jQuery in that file
 import $ from 'jquery';
 
-// An example of how you tell webpack to use a CSS (SCSS) file
 import './css/base.scss';
 
-// An example of how you tell webpack to use an image (also need to link to it in the index.html)
-import './images/turing-logo.png'
+import User from "./User";
+import Manager from "./Manager";
+import Roombooker from "./Rooms";
+
+
+import './images/raven-logo-2.jpg';
+import './images/pngtree-feather.png';
+import './images/disco-lounge.png';
+import './images/disco-gardens.png';
+import './images/disco-pool.png'
 
 console.log('This is the JavaScript entry file - your code begins here.');
+
+let today = new Date();
+let day = today.getDate().toString().padStart(2, '0');
+let month = (today.getMonth() + 1).toString().padStart(2, '0');
+let year = today.getFullYear().toString();
+today = `${year}/${month}/${day}`;
+
+console.log(today);
+
+const userFetch = fetch('https://fe-apps.herokuapp.com/api/v1/overlook/1904/users/users')
+  .then(response => response.json())
+  .then(data => {
+    return (data.users)})
+  .catch(data => console.log('Fetch error - user data. User may not be defined.', data));
+
+const roomFetch = fetch('https://fe-apps.herokuapp.com/api/v1/overlook/1904/rooms/rooms')
+  .then(response => response.json())
+  .then(data => {
+    return (data.rooms)})
+  .catch(data => console.log('Fetch error - room data. Room info may not be defined.', data));
+
+
+const bookingFetch = fetch('https://fe-apps.herokuapp.com/api/v1/overlook/1904/bookings/bookings')
+  .then(response => response.json())
+  .then(data => {
+    return (data.bookings)})
+  .catch(data => console.log('Fetch error - booking data. Booking info may not be defined.', data));
+
+  $('#logInButton').click((event) => {
+
+    Promise.all([userFetch, roomFetch, bookingFetch]).then((requiredData) => {
+      const users = requiredData[0];
+      const rooms = requiredData[1];
+      const bookings = requiredData[2];
+    if ($('#username-input').val().includes('manager') ) {
+      $('#managerPage').removeClass('hidden');
+      const manager = new Manager(users, bookings, rooms, $('#username-input').val());
+      hydrateManagerPage(manager, today);
+      $('#managerFormButton').click((event) => {
+        $('#managerSearchBox').removeClass('hidden');
+        $('#managerCenterpiece').addClass('hidden');
+      });
+      $('#managerSearchButton').click((event) => {
+        $('#newUserInfo').removeClass('hidden');
+        const newUserId = manager.getUserIdByName($('#newUserInput').val());
+        hydrateManagerSearch(manager, today, newUserId);
+      });
+    }
+    if ($('#username-input').val().includes('customer') ) {
+      $('#userPage').removeClass('hidden');
+      const user = new User(users, bookings, rooms, $('#username-input').val());
+      hydrateUserPage(user, today);
+    }
+    $('#loginPage').addClass('hidden');
+    })
+  });
+
```

Comment:
So I think it might be helpful if I were able to review all the things that this function does and that way, it might be possible to split some of it up into different functions that can simply be invoked when each if statement happens. I'm also curious if the promise.all needs to happen within the click event listener. It could also be done and then once all the fetch requests are done, then the event listener is added. That's a tangent though.

1. [lines 46-52] When the button is clicked, the fetch requests are all re-assigned to global variables once that's happened (within the promise.all()). -> if the button is hit super quickly, it might move on before the promise.all() is complete so I might suggest a .then() on line 51.5 to make sure the promise.all() is done before moving on to the if conditionals.
2. [lines 52-65] If manager is included in the username input, then switch to the manager page, instantiate a manager, populate the page with information, add an event listener on the form, and add an event listener on the search button. -> I feel like the if conditional needs to be there, but you could put lines 53-64 into one function called like 'switchToManagerPage' that could hold all of that in a separate function declaration later in the file.
3. [lines 66-73] If customer is included in the username input, then switch to the user page, instantiate a new user, and then populate the page with information. At the end no matter what, hide the login page. -> Similar to the 'switchToManagerPage' I think you could create a 'switchToUserPage' function that would include lines 67-69 and then that would just invoke it within the conditional logic. 

Overall this looks great and breaking out some of the functionality into smaller functions could be **fingers crossed** hopefully pretty simple. I would keep the conditional logic but just have baby functions that do all the things within the conditional logic. 

Hope this helps!